main.py

```diff
@@ -0,0 +1,84 @@
+from sklearn import tree, model_selection, naive_bayes
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn import ensemble
+from sklearn import preprocessing
+import xgboost as xgb
+from sklearn.model_selection import train_test_split, ShuffleSplit, LearningCurveDisplay
+import joblib
+import pandas as pd
+
+filename = 'weights/titanic_survivor_grad_boost_classifier.pkl'
+
+
+def convert_csv_to_arrays(path_to_csv: str, train: bool):
+    original_df = pd.read_csv(path_to_csv, sep=',', header=0)\
+        .drop('Ticket', axis=1)
+    original_df.interpolate(inplace=True)
+
+    if train:
+        out_df = original_df['Survived'].copy()
+        in_df = original_df.drop('Survived', axis=1)
+    else:
+        in_df = original_df.copy()
+
+    in_df = in_df
+    in_df['Sex'] = in_df['Sex'].map(lambda x: 1 if x == 'male' else 0)
+    label_encoder = preprocessing.LabelEncoder()
+    names_encoder = preprocessing.OrdinalEncoder()
+    cabin_encoder = preprocessing.LabelEncoder()
+
+    in_df['Embarked'] = label_encoder.fit_transform(in_df['Embarked'])
+    in_df['Name'] = names_encoder.fit_transform(in_df['Name'].values.reshape(-1, 1))
+    in_df['Cabin'] = cabin_encoder.fit_transform(in_df['Cabin'].values.reshape(-1, 1))
+    print(in_df.head(5))
+    X = in_df.drop('PassengerId', axis=1).to_numpy()
+    if train:
+        y = out_df.to_numpy()
+        return y, X
+    return X, original_df['PassengerId']
```

Comment:
Returning variables in a different order is confusing here. Can you please return variables in the same order (`X, y` or `y, X`) for both train and test?